### PR TITLE
chore: Update node version in CI and Volta

### PR DESCRIPTION
**Pull request by MehulKChaudhari**
Date: 6/11/2024

Developer Name: @MehulKChaudhari 

---

## Description

Used the latest Node version instead of `18` and add timeout in CI workflow

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,12 +23,12 @@ jobs:
...
         uses: pnpm/action-setup@v2
         with:
-          version: 8
+          version: 9.12.3
 
       - name: Setup Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 18
+          node-version: 22.11.0
```
Comment (prakashchoudhary07) [inline .github/workflows/ci.yml]: Can we just write 22? 
What difference will it make?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ on:
 jobs:
   lint_test:
     runs-on: ubuntu-latest
+    timeout-minutes: 10
     steps:
       - name: Checkout
         uses: actions/checkout@v3
```
Comment (prakashchoudhary07) [inline .github/workflows/ci.yml]: Is this the latest?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,12 +23,12 @@ jobs:
       - name: Install pnpm
         uses: pnpm/action-setup@v2
```
Comment (prakashchoudhary07) [inline .github/workflows/ci.yml]: Re: is this the latest one?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,6 +13,7 @@ on:
 jobs:
   lint_test:
     runs-on: ubuntu-latest
+    timeout-minutes: 10
```
Comment (prakashchoudhary07) [inline .github/workflows/ci.yml]: Will it take 10 mins?
Lets reduce it, and at later point when we need more time we can always update
I think 5 should be good enough for now?
What do you think. 
How much time does the build takes currently

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (MehulKChaudhari) [reply .github/workflows/ci.yml]: Yup 5 sounds good, its taking around 2m45secs

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MehulKChaudhari) [reply .github/workflows/ci.yml]: Yes

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MehulKChaudhari) [reply .github/workflows/ci.yml]: updated

File: .github/workflows/ci.yml (hunk 3 above)
Comment (MehulKChaudhari) [reply .github/workflows/ci.yml]: Updating

**Review by prakashchoudhary07 — DISMISSED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**